pkgs/development/compilers/haxe/haxe-packages.nix

```diff
@@ -0,0 +1,274 @@
+    name = "${libname}-${version}";
+
+    src = fetchzip {
+      inherit name;
+      url = "http://lib.haxe.org/files/3.0/${withCommas name}.zip";
+      sha256 = "1agd3063fpy5gw15l91l4xqiv2jfww1bfai5vqn83p5nsrdqvvq2";
+    };
+
+    buildCommand = ''
+      mkdir -p "$out/lib/haxe/${libname}/${withCommas version}"
+      echo -n "${version}" > $out/lib/haxe/${libname}/.current
+      cp -dpR $src/* "$out/lib/haxe/${libname}/${withCommas version}/"
+    '';
+
+    meta = with stdenv.lib; {
+      homepage = "http://lib.haxe.org/p/${libname}";
+      description = "Support library for the Java backend of the Haxe compiler";
+      license = licenses.bsd2;
+      platforms = platforms.linux;
+    };
+  };
+
+
+  hxcs = stdenv.mkDerivation rec {
+    libname = "hxcs";
+    version = "3.4.0";
+    name = "${libname}-${version}";
+
+    src = fetchzip {
+      inherit name;
+      url = "http://lib.haxe.org/files/3.0/${withCommas name}.zip";
+      sha256 = "0f5vgp2kqnpsbbkn2wdxmjf7xkl0qhk9lgl9kb8d5wdy89nac6q6"; #"1lchs080ym65p6yis7j38ba2qc1l6ax2c0rsawkd07jc3mnkc1d4";
+      stripRoot = false;
+    };
+
+    buildCommand = ''
+      mkdir -p "$out/lib/haxe/${libname}/${withCommas version}"
+      echo -n "${version}" > $out/lib/haxe/${libname}/.current
+      cp -dpR $src/* "$out/lib/haxe/${libname}/${withCommas version}/"
+    '';
+
+    meta = with stdenv.lib; {
+      homepage = "http://lib.haxe.org/p/${libname}";
+      description = "Support library for the C# backend of the Haxe compiler";
+      license = licenses.bsd2;
+      platforms = platforms.linux;
+    };
+  };
+
+
+  actuate = stdenv.mkDerivation rec {
+    libname = "actuate";
+    version = "1.8.7";
+    name = "${libname}-${version}";
+
+    src = fetchzip {
+      inherit name;
+      url = "http://lib.haxe.org/files/3.0/${withCommas name}.zip";
+      sha256 = "02ma4cf8dhn82bk8ybin24jw9fwwyjn7l07vk4gbkhlngb1qibsy";
+    };
+
+    buildCommand = ''
+      mkdir -p "$out/lib/haxe/${libname}/${withCommas version}"
+      echo -n "${version}" > $out/lib/haxe/${libname}/.current
+      cp -dpR $src/* "$out/lib/haxe/${libname}/${withCommas version}/"
+    '';
+
+    meta = with stdenv.lib; {
+      homepage = "http://lib.haxe.org/p/${libname}";
+      description = ''Actuate is a fast and flexible tween library that uses a jQuery-style "chaining" syntax'';
+      license = licenses.bsd2;
+      platforms = platforms.linux;
+    };
+  };
+
+
+  lime = stdenv.mkDerivation rec {
+    libname = "lime";
+    version = "5.0.0";
+    name = "${libname}-${version}";
+
+    src = fetchzip {
+      inherit name;
+      url = "http://lib.haxe.org/files/3.0/${withCommas name}.zip";
+      sha256 = "0b6phkjfhh34dn8x8dli4100aj9bx705x1l43czqdawcvr3mw61i";
+    };
+
+    buildCommand = ''
+      mkdir -p "$out/lib/haxe/${libname}/${withCommas version}"
+      echo -n "${version}" > $out/lib/haxe/${libname}/.current
+      cp -dpR $src/* "$out/lib/haxe/${libname}/${withCommas version}/"
```

Comment:
Some abstraction could be helpful, as all libraries seems to be build the same way.